fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1158,18 +1162,208 @@ void partitionedJoinedIndexTest() {
+
+            // Get 18 results and continuation
+            List<IndexEntry> entries = indexEntryCursor.asList().join();
+            assertEquals(18, entries.size());
+            RecordCursorResult<IndexEntry> lastResult = indexEntryCursor.onNext().get();
+            assertEquals(RecordCursor.NoNextReason.RETURN_LIMIT_REACHED, lastResult.getNoNextReason());
+            LuceneContinuationProto.LuceneIndexContinuation parsed = LuceneContinuationProto.LuceneIndexContinuation.parseFrom(lastResult.getContinuation().toBytes());
+            // we stopped in partition 0
+            assertEquals(0, parsed.getPartitionId());
+            final Set<Tuple> expectedKeys;
+            if (sortType == SortType.ASCENDING || sortType == SortType.UNSORTED) {
+                expectedKeys = Set.copyOf(primaryKeys.subList(0, 18));
+            } else {
+                expectedKeys = Set.copyOf(primaryKeys.subList(7, 25));
+            }
+
+            assertEquals(expectedKeys, entries.stream().map(IndexEntry::getPrimaryKey).collect(Collectors.toSet()));
+            continuation = lastResult.getContinuation().toBytes();
+        }
+
+        // run re-partitioning
+        explicitMergeIndex(index, contextProps, schemaSetup);
+        // now there should be 4 partitions:
+        //  partition 0: with docs 18 - 24
+        //  partition 3: with docs 12 - 17
+        //  partition 2: with docs  6 - 11
+        //  partition 1: with docs  0 - 5
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            validateDocsInPartition(index, 0, groupTuple, Set.copyOf(primaryKeys.subList(18, 25)), luceneSearch);
+            validateDocsInPartition(index, 3, groupTuple, Set.copyOf(primaryKeys.subList(12, 18)), luceneSearch);
+            validateDocsInPartition(index, 2, groupTuple, Set.copyOf(primaryKeys.subList(6, 12)), luceneSearch);
+            validateDocsInPartition(index, 1, groupTuple, Set.copyOf(primaryKeys.subList(0, 6)), luceneSearch);
+        }
+
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+
+            RecordCursor<IndexEntry> indexEntryCursor = recordStore.scanIndex(index, scanQuery, continuation, ExecuteProperties.newBuilder().setReturnedRowLimit(11).build().asScanProperties(false));
+
+            // so now we should get the remaining docs
+            List<IndexEntry> entries = indexEntryCursor.asList().join();
+            final Set<Tuple> expectedKeys;
+            final RecordCursor.NoNextReason noNextReason;
+            final RecordCursorResult<IndexEntry> lastResult = indexEntryCursor.onNext().get();
+            final int expectedCount;
+            if (sortType == SortType.ASCENDING) {
+                expectedKeys = Set.copyOf(primaryKeys.subList(18, 25));
+                expectedCount = 7;
+                noNextReason = RecordCursor.NoNextReason.SOURCE_EXHAUSTED;
```

Comment:
Due to the fact that you just start with `0`, this ends up continuing in the same partition id as it was before. It would probably be good if this continued in partition 3.